packages/amplify_api_rest_plugin_interface/lib/src/RestOptions.dart

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:flutter/foundation.dart';
+import 'dart:typed_data';
+
+class RestOptions {
+  String apiName;
+  String path;
+  Uint8List body;
```

Comment:
not `ByteData`?

Reply:
I believe it's easy to convert, if needed:
```
Uint8List bytes = ...;
var blob = ByteData.sublistView(bytes);
```

Reply:
Uint8List is the data format for sending byte data across platform channels.  Using ByteArray would require a conversion from Uint8List when receiving and sending information over platform channels.  That's the main reason.  Is ByteData a better data format for the users?  If so we could convert to ByteData for user access though perhaps it's better to return a lower level data object? @Amplifiyer @haverchuck

Reply:
>Uint8List is the data format for sending byte data across platform channels.

Are we sure about this? From this [interface](https://github.com/flutter/flutter/blob/de448c81ed1b5992c485a86e1de4624d02027981/packages/flutter/lib/src/services/binary_messenger.dart#L34) it looks like flutter uses `ByteData`

From a quick search, it seems `ByteData` is a popular format representing binary data?

Reply:
Hmm weird .. from the dart docs on platform code I got this:

https://flutter.dev/docs/development/platform-integration/platform-channels
![image](https://user-images.githubusercontent.com/11478656/102837877-aca99780-43b1-11eb-975a-efdeddf503d4.png)

So I drew the conclusion that Uint8List was the popular format for binary data.  Perhaps their docs are wrong then if this interface is using ByteArray?  I just thought I would have gotten a class cast exception error or something as I assumed the format was Uint8List when writing the dart code. 

@Amplifiyer

Reply:
hmm it seems ByteData is a higher level abstraction? I'm fine using Uint8List since it can be easily converted into a ByteData as @haverchuck mentioned.